indico/web/client/js/react/components/pictures/PictureArea.jsx

```diff
@@ -0,0 +1,238 @@
+// LICENSE file for more details.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {
+  Button,
+  Card,
+  Divider,
+  Grid,
+  Header,
+  Segment,
+  Image,
+  Icon,
+  Popup,
+  Progress,
+  Message,
+} from 'semantic-ui-react';
+
+import {Translate} from 'indico/react/i18n';
+
+import {dropzoneShape, fileActionShape} from '../files/props';
+
+import './Picture.module.scss';
+
+export function PictureArea({
+  dropzone: {getRootProps, getInputProps, isDragActive, open: openUploadDialog, fileRejections},
+  capture: {pictureCamera: PictureCamera, onOpenCameraDialog, isCameraActive, isCameraAllowed},
+  cropper: {pictureCropper: PictureCropper, isEditActive},
+  picture,
+  dragText,
+  descriptionText,
+  picturePlaceholder,
+  pictureAction,
+  picturePreview,
+  customFileRejections,
+}) {
+  const uploadButtonIcon = picture ? 'exchange' : 'upload';
+  const uploadButtonText = picture
+    ? Translate.string('Choose a new picture from your computer')
+    : Translate.string('Choose from your computer');
+  const captureButtonIcon = 'camera';
+  const captureButtonText = picture
+    ? Translate.string('Take a new picture')
+    : Translate.string('Take a picture');
+  const dropzoneDisabled = openUploadDialog === null;
+
+  return (
+    <div {...getRootProps()} styleName="dropzone-area">
+      <input {...getInputProps()} />
+      <Segment textAlign="center" placeholder>
+        {picture && picture.imageSrc && isEditActive && <PictureCropper />}
+        {isCameraActive && <PictureCamera />}
+        {!isEditActive && !isCameraActive && (
+          <Grid celled="internally">
+            <Grid.Row columns={picture ? 2 : 1}>
+              {!isDragActive && picture && (
+                <Grid.Column width={10} verticalAlign="middle">
+                  <Card styleName="picture-card" key={picture.filename} centered>
+                    <Image src={picturePreview()} size="large" rounded />
+                    {picture.upload && !picture.upload.finished && (
+                      <Progress
+                        percent={picture.upload.progress}
+                        error={picture.upload.failed}
+                        size="tiny"
+                        active
+                        color="blue"
+                      />
+                    )}
+                    {pictureAction && (
+                      <Icon
+                        styleName="action"
+                        name={pictureAction.icon}
+                        color={pictureAction.color}
+                        onClick={() => pictureAction.onClick(picture)}
+                      />
+                    )}
+                  </Card>
+                </Grid.Column>
+              )}
+              <Grid.Column verticalAlign="middle" width={!isDragActive && picture ? 6 : 16}>
+                {!picture && (
+                  <Icon name={picturePlaceholder} size="massive" styleName="picture-placeholder" />
```

Comment:
I would get rid of this. It just wastes lots of space (the upload widgets are already huge enough...) and has no real benefit

Reply:
okay